### PR TITLE
DNM: mon/OSDMonitor: Add cluster address in stream lines

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -1126,31 +1126,43 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
...
-       i != pending_inc.new_weight.end();
-       ++i) {
-    if (i->second == CEPH_OSD_OUT) {
-      dout(2) << " osd." << i->first << " OUT" << dendl;
-    } else if (i->second == CEPH_OSD_IN) {
-      dout(2) << " osd." << i->first << " IN" << dendl;
-    } else {
-      dout(2) << " osd." << i->first << " WEIGHT " << hex << i->second << dec << dendl;
+  {
+    OSDMap tmp;
```
Comment (jecluis) [inline src/mon/OSDMonitor.cc]: You are creating a new variable named `tmp` within this block that is not initialized. All your outputs below relying on `tmp.get_cluster_addr()` should not be returning what you want. This information should already be available in the original `tmp` map, initialized at the beginning of the function.

File: src/mon/OSDMonitor.cc [hunk 2]
```diff
@@ -1126,31 +1126,43 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
...
+    // tell me about it
+    for (auto i = pending_inc.new_state.begin();
+      i != pending_inc.new_state.end(); ++i) {
+      int s = i->second ? i->second : CEPH_OSD_UP;
+      int32_t osd = i->first;
+      const entity_addr_t cluster = tmp.get_cluster_addr(osd);
+      if (s & CEPH_OSD_UP) 
+        dout(2) << "cluster " << cluster << " "
```
Comment (jecluis) [inline src/mon/OSDMonitor.cc]: Calling this `cluster` may be confusing - I would instead go with `cluster_addr`. Also, I would suggest moving the cluster addr to after `osd`. E.g.,

`dout(2) << "osd." << osd << " cluster_addr " << cluster_addr << " ..." << dendl;`

File: src/mon/OSDMonitor.cc (hunk 1 above)
Comment (shinobu-x) [reply src/mon/OSDMonitor.cc]: Yeah, that's my bad :)

File: src/mon/OSDMonitor.cc (hunk 2 above)
Comment (shinobu-x) [reply src/mon/OSDMonitor.cc]: Thanks for your suggestion!

**Comment by shinobu-x** [comment]
test this please

**Pull request closed without merging**